### ISSUE TITLE
Document-relative urls like '?a=b'

### ISSUE THREAD

**Issue by xppt** [aio-libs/yarl#99]
`URL.with_query()` currently sets new instance path to '/' if it was empty.
```
>> URL.build(query={'a': '1'})
<< URL('/?a=1')
```
It prevents building relative url with only query seqment, like '?a=b'.
On the other hand, urlunsplit is working fine in this case:
```
urlunsplit(('', '', '', 'ab=3', None))
'?ab=3'
```
I think yarl should not replace empty path in `with_query` implementation.

**Comment by asvetlov** [comment]
Make sense.
Would you provide a PR?

**Comment by asvetlov** [comment]
Just curious: where do you use such URLs?

**Comment by xppt** [comment]
Added PR.
I've got some service, which parameter is an external document URL. In its page I need to offer user a few links to the same service, but with different parameter value. And I can't use absolute path URLs because this service is available with a few different host names and paths (as a part of separate projects).
It seems like a very useful URL option.

**Comment by asvetlov** [comment]
Fixed by #100
